src/material/button/button.scss

```diff
@@ -6,15 +6,144 @@
+      // these slots ourselves.
+      pressed-state-layer-color: null,
+      pressed-state-layer-opacity: null,
+      // MDC structures its icons differently, so we emit these slots ourselves.
+      with-icon-icon-size: null,
+      with-icon-icon-color: null,
+      with-icon-hover-icon-color: null,
+      with-icon-focus-icon-color: null,
+      with-icon-pressed-icon-color: null,
+      with-icon-disabled-icon-color: null,
+    )));
+
+    // Add default values for MDC text button tokens that aren't outputted by the theming API.
+    @include mdc-button-text-theme.theme(tokens-mdc-button-text.get-unthemable-tokens());
+
+    @include token-utils.use-tokens(tokens-mdc-button-text.$prefix, $mdc-button-text-slots) {
+      // Angular does not use the `:focus` pseudo-class to display focus styles. Instead we emit the
+      // focus-related slots under the `.cdk-keyboard-focused` class
+      &.cdk-keyboard-focused .mdc-button__ripple::before {
+        @include token-utils.create-token-slot(opacity, focus-state-layer-opacity);
+        @include token-utils.create-token-slot(background-color, focus-state-layer-color);
+      }
+
+      // We prefer to have the pressed state on the `::before` element rather than `::after` where
+      // MDC adds it, so we emit these slots ourselves.
+      &:active .mdc-button__ripple::before {
+        @include token-utils.create-token-slot(background-color, pressed-state-layer-color);
+        @include token-utils.create-token-slot(opacity, pressed-state-layer-opacity);
+      }
+
+      // We structure our icons differently than MDC in order to take advantage of the mat-icon
+      // component. Therefore we need to emit slots related to the text button icons ourselves.
+      & > .mat-icon {
+        @include token-utils.create-token-slot(font-size, with-icon-icon-size);
+        @include token-utils.create-token-slot(color, with-icon-icon-color);
+      }
+
+      &:hover > .mat-icon {
+        @include token-utils.create-token-slot(color, with-icon-hover-icon-color);
+      }
+
+      &.cdk-keyboard-focused > .mat-icon {
+        @include token-utils.create-token-slot(color, with-icon-focus-icon-color);
+      }
+
+      &:active > .mat-icon {
+        @include token-utils.create-token-slot(color, with-icon-pressed-icon-color);
+      }
 
+      // MDC targets disabled styles at the `:disabled` pseudo-class which doesn't work for anchors,
+      // Therefore we emit these slots ourselves targeted at `[disabled]` instead.
+      &[disabled] {
+        @include token-utils.create-token-slot(color, disabled-label-text-color);
+
+        & > .mat-icon {
+          @include token-utils.create-token-slot(color, with-icon-disabled-icon-color);
+        }
+      }
+
+      // Make the ripple the same color as the button text.
+      .mat-ripple-element {
+        @include token-utils.create-token-slot(background-color, label-text-color);
+      }
+    }
+  }
+}
+
+.mat-mdc-button {
```

Comment:
Couldn't this be combined with the `.mat-mdc-button` selector above?

Reply:
It could, though eventually we'll add the token-related styles for the other variants above, and I wanted to keep the styles that were not token-related separate. Long term I'd like to try to reduce the amount of extra styles we need on top of MDC's styles, so this is kind of like the section of stuff I'd like to focus on eliminating by getting more in sync with MDC